### PR TITLE
modify mongodb force_delete

### PR DESCRIPTION
**Pull request by cyberHermanwang**
tencentcloud_mongodb_sharding_instance.mongodb_postpaid-sharding-keep: Destruction complete after 1m16s
tencentcloud_security_group.foo: Destroying... [id=sg-bnzg8an9]
tencentcloud_subnet.foo: Destroying... [id=subnet-lvj0mqwg]
tencentcloud_security_group.foo: Destruction complete after 1s
tencentcloud_subnet.foo: Destruction complete after 2s
tencentcloud_vpc.foo: Destroying... [id=vpc-b5zhxz0j]
tencentcloud_vpc.foo: Destruction complete after 2s

Destroy complete! Resources: 4 destroyed.

### CODE REVIEW

**Comment by cyberHermanwang** [comment]
tencentcloud_mongodb_sharding_instance.mongodb_postpaid-sharding-keep: Destruction complete after 1m16s
tencentcloud_security_group.foo: Destroying... [id=sg-bnzg8an9]
tencentcloud_subnet.foo: Destroying... [id=subnet-lvj0mqwg]
tencentcloud_security_group.foo: Destruction complete after 1s
tencentcloud_subnet.foo: Destruction complete after 2s
tencentcloud_vpc.foo: Destroying... [id=vpc-b5zhxz0j]
tencentcloud_vpc.foo: Destruction complete after 2s

Destroy complete! Resources: 4 destroyed.



tencentcloud_mongodb_instance.mongodb_postpaid-keep: Destroying... [id=cmgo-89j2ehcb]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 10s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 20s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 30s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 40s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 50s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Still destroying... [id=cmgo-89j2ehcb, 1m0s elapsed]
tencentcloud_mongodb_instance.mongodb_postpaid-keep: Destruction complete after 1m4s
tencentcloud_security_group.foo: Destroying... [id=sg-e9hquep5]
tencentcloud_subnet.foo: Destroying... [id=subnet-0v6bm9mg]
tencentcloud_security_group.foo: Destruction complete after 1s
tencentcloud_subnet.foo: Destruction complete after 1s
tencentcloud_vpc.foo: Destroying... [id=vpc-gikeav4f]
tencentcloud_vpc.foo: Destruction complete after 2s

Destroy complete! Resources: 4 destroyed.

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 1]
```diff
@@ -511,6 +511,24 @@ func resourceTencentCloudMongodbInstanceDelete(d *schema.ResourceData, meta inte
 	if err != nil {
 		return err
 	}
-
+	err = mongodbService.OfflineIsolatedDBInstance(ctx, instanceId)
+	if err != nil {
+		log.Printf("[CRITAL]%s mongodb %s fail, reason:%s\n", logId, "OfflineIsolatedDBInstance", err.Error())
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_mongodb_instance.go:516]: printf最后会加换行，多余了

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 2]
```diff
@@ -511,6 +511,24 @@ func resourceTencentCloudMongodbInstanceDelete(d *schema.ResourceData, meta inte
...
+		return err
+	}
+	//describe and check not exist
+	err = resource.Retry(5*readRetryTimeout, func() *resource.RetryError {
+		instance, _, errRet := mongodbService.DescribeInstanceById(ctx, instanceId)
+		if errRet != nil {
+			return retryError(errRet, InternalError)
+		}
+		if instance == nil {
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_mongodb_instance.go:525]: 用第二个返回值判断，没必要判空

**Review by likexian — APPROVED** [review]

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 3]
```diff
@@ -313,3 +313,43 @@ func (me *MongodbService) DescribeAsyncRequestInfo(ctx context.Context, asyncId
...
+func (me *MongodbService) OfflineIsolatedDBInstance(ctx context.Context, instanceId string) (errRet error) {
+	logId := getLogId(ctx)
+	request := mongodb.NewOfflineIsolatedDBInstanceRequest()
+	request.InstanceId = &instanceId
+	defer func() {
+		if errRet != nil {
+			log.Printf("[CRITAL]%s api[%s] fail,reason[%s]", logId, request.GetAction(), errRet.Error())
+		}
+	}()
+	var response *mongodb.OfflineIsolatedDBInstanceResponse
```
Comment (brickzzhang) [inline tencentcloud/service_tencentcloud_mongodb.go:326]: var err error，少一个变量

File: tencentcloud/service_tencentcloud_mongodb.go [hunk 4]
```diff
@@ -313,3 +313,43 @@ func (me *MongodbService) DescribeAsyncRequestInfo(ctx context.Context, asyncId
...
+	err := resource.Retry(writeRetryTimeout, func() *resource.RetryError {
+		ratelimit.Check(request.GetAction())
+		result, e := me.client.UseMongodbClient().OfflineIsolatedDBInstance(request)
```
Comment (brickzzhang) [inline tencentcloud/service_tencentcloud_mongodb.go:329]: 用response就行，没必要定义result

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 5]
```diff
@@ -511,6 +511,24 @@ func resourceTencentCloudMongodbInstanceDelete(d *schema.ResourceData, meta inte
...
+			return nil
+		}
+		return resource.NonRetryableError(fmt.Errorf("after OfflineIsolatedDBInstance mongodb Status is %d", *instance.Status))
```
Comment (crab21) [inline tencentcloud/resource_tc_mongodb_instance.go:528]: instance为nil时，painc了。

File: tencentcloud/resource_tc_mongodb_instance.go [hunk 6]
```diff
@@ -511,6 +511,24 @@ func resourceTencentCloudMongodbInstanceDelete(d *schema.ResourceData, meta inte
...
 	}
-
+	err = mongodbService.OfflineIsolatedDBInstance(ctx, instanceId)
+	if err != nil {
+		log.Printf("[CRITAL]%s mongodb %s fail, reason:%s\n", logId, "OfflineIsolatedDBInstance", err.Error())
+		return err
+	}
+	//describe and check not exist
+	err = resource.Retry(5*readRetryTimeout, func() *resource.RetryError {
+		instance, _, errRet := mongodbService.DescribeInstanceById(ctx, instanceId)
```
Comment (crab21) [inline tencentcloud/resource_tc_mongodb_instance.go:521]: 这个DescribeInstanceById查询里面重试60分钟，上面又是15分钟，这个是不是最差情况得等好久～ 多个小时

**Review by gailwang — APPROVED** [review]

**Pull request merged**